core/lib/spree/core/controller_helpers/auth.rb

```diff
@@ -24,10 +24,12 @@ def redirect_back_or_default(default)
           session['spree_user_return_to'] = nil
         end
 
-        def set_guest_token
-          if cookies.signed[:guest_token].blank?
-            cookies.permanent.signed[:guest_token] = { value: generate_guest_token, httponly: true }
-          end
+        def set_token
+          cookies.permanent.signed[:token] ||= cookies.delete(:guest_token)
+          cookies.permanent.signed[:token] ||= {
```

Comment:
@damianlegawiec @bbonislawski What do you think about something like this?e

Reply:
Awesome 👍